Apps/GatewayApi/src/Services/DependentService.cs

```diff
@@ -192,6 +193,43 @@ public RequestResult<IEnumerable<DependentModel>> GetDependents(string hdId, int
             return result;
         }
 
+        /// <inheritdoc/>
+        public RequestResult<IEnumerable<GetDependentResponse>> GetDependents(string fromDateUtc, string? toDateUtc, int page = 0, int pageSize = 5000)
+        {
+            // Page size max is 5000
+            pageSize = pageSize <= 5000 ? pageSize : 5000;
+
+            DateTime startDate = DateTime.Parse(fromDateUtc, CultureInfo.InvariantCulture);
+            startDate = DateTime.SpecifyKind(startDate, DateTimeKind.Utc);
+            DateTime endDate = DateTime.Parse(toDateUtc, CultureInfo.InvariantCulture);
+            endDate = DateTime.SpecifyKind(endDate, DateTimeKind.Utc);
+
+            // Get Dependents from database
+            int offset = page * pageSize;
+            DbResult<IEnumerable<ResourceDelegate>> dbResourceDelegates = this.resourceDelegateDelegate.Get(
+                startDate,
+                endDate,
+                offset,
+                pageSize);
+
+            if (dbResourceDelegates.Status != DbStatusCode.Read)
+            {
+                return RequestResultFactory.ServiceError<IEnumerable<GetDependentResponse>>(ErrorType.CommunicationInternal, ServiceType.Database, dbResourceDelegates.Message);
+            }
+
+            IDictionary<string, IEnumerable<ResourceDelegate>> resourceDelegatesByProfileId =
+                dbResourceDelegates.Payload.GroupBy(g => g.ProfileHdid).ToDictionary(g => g.Key, g => g.AsEnumerable());
+            IEnumerable<GetDependentResponse> dependentResponses = resourceDelegatesByProfileId.Select(
+                    keyValuePair => new GetDependentResponse
+                    {
+                        DelegateId = keyValuePair.Key,
+                        DependentRecords = this.autoMapper.Map<IEnumerable<DependentRecord>>(keyValuePair.Value),
+                    })
+                .ToList();
```

Comment:
grrr.. ToList()...

Reply:
I tried removing the ToList() but then I get a "Possible Multiple Enumeration" warning on the next line: 
return RequestResultFactory.Success(dependentResponses, dependentResponses.Count(), page, pageSize);

What is the solution to that issue?

Reply:
I tried removing the ToList() but then I get a "Possible Multiple Enumeration" warning on the next line: 
return RequestResultFactory.Success(dependentResponses, dependentResponses.Count(), page, pageSize);

What is the solution to that issue?

Reply:
You can enumerate with ToArray(), but it's not a big deal